src/main/java/com/google/cloud/tools/app/InvalidDirectoryException.java

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.tools.app;
+
+/**
+ * Signals an invalid directory.
+ */
+public class InvalidDirectoryException extends RuntimeException {
```

Comment:
At the beginning, I had a special treatment for these in the clients, but that was a long way back, so I'll get rid of them and use IAEs.